src/shared/calendarspec.c

```diff
@@ -1235,6 +1235,15 @@ static bool matches_weekday(int weekdays_bits, const struct tm *tm, bool utc) {
         return (weekdays_bits & (1 << k));
 }
 
+static int compare_tm(const struct tm *t1, const struct tm *t2) {
+        if (t1->tm_year != t2->tm_year) return t1->tm_year - t2->tm_year;
+        if (t1->tm_mon  != t2->tm_mon)  return t1->tm_mon  - t2->tm_mon;
+        if (t1->tm_mday != t2->tm_mday) return t1->tm_mday - t2->tm_mday;
+        if (t1->tm_hour != t2->tm_hour) return t1->tm_hour - t2->tm_hour;
+        if (t1->tm_min  != t2->tm_min)  return t1->tm_min  - t2->tm_min;
+        return t1->tm_sec - t2->tm_sec;
```

Comment:
please use the CMP() macro, it's guaranteed to handle overflows correctly.
also, please follow coding style, i.e. do not cram if + return in the same line